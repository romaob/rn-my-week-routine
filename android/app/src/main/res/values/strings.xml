<resources>
    <string name="app_name">MyWeekRoutine</string>
</resources>
