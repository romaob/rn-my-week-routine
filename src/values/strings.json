{
    "app_name": {
        "en": "My Week Routine",
        "es": "Mi Rutina Semanal",
        "pt": "Minha Rotina Semanal"
    },
    "screen_routine": {
        "en": "Routine",
        "es": "Rutina",
        "pt": "Rotina"
    },
    "screen_events": {
        "en": "My Events",
        "es": "Mis Eventos",
        "pt": "Meus Eventos"
    },
    "screen_settings": {
        "en": "Settings",
        "es": "Ajustes",
        "pt": "Configurações"
    },
    "screen_about": {
        "en": "About",
        "es": "Acerca de",
        "pt": "Sobre"
    },
    "lang_en": {
        "en": "English",
        "es": "English",
        "pt": "English"
    },
    "lang_es": {
        "en": "Español",
        "es": "Español",
        "pt": "Español"
    },
    "lang_pt": {
        "en": "Português",
        "es": "Português",
        "pt": "Português"
    },
    "test": {
        "en": "Testing the app strings",
        "es": "Probando las cadenas de la aplicación",
        "pt": "Testando as strings do app"
    },
    "week_day_mon": {
        "en": "Monday",
        "es": "Lunes",
        "pt": "Segunda"
    },
    "week_day_tue": {
        "en": "Tuesday",
        "es": "Martes",
        "pt": "Terça"
    },
    "week_day_wed": {
        "en": "Wednesday",
        "es": "Miércoles",
        "pt": "Quarta"
    },
    "week_day_thu": {
        "en": "Thursday",
        "es": "Jueves",
        "pt": "Quinta"
    },
    "week_day_fri": {
        "en": "Friday",
        "es": "Viernes",
        "pt": "Sexta"
    },
    "week_day_sat": {
        "en": "Saturday",
        "es": "Sábado",
        "pt": "Sábado"
    },
    "week_day_sun": {
        "en": "Sunday",
        "es": "Domingo",
        "pt": "Domingo"
    },
    "save": {
        "en": "Save",
        "es": "Guardar",
        "pt": "Salvar"
    },
    "delete": {
        "en": "Delete",
        "es": "Eliminar",
        "pt": "Excluir"
    },
    "cancel": {
        "en": "Cancel",
        "es": "Cancelar",
        "pt": "Cancelar"
    },
    "ok": {
        "en": "OK",
        "es": "OK",
        "pt": "OK"
    },
    "routine_new": {
        "en": "New Routine",
        "es": "Nueva Rutina",
        "pt": "Nova Rotina"
    },
    "routine_name": {
        "en": "Routine Name",
        "es": "Nombre de la Rutina",
        "pt": "Nome da Rotina"
    },
    "routine_description": {
        "en": "Routine Description",
        "es": "Descripción de la Rutina",
        "pt": "Descrição da Rotina"
    },
    "routine_days_select": {
        "en": "On which days do you want to do this routine?",
        "es": "¿En qué días quieres hacer esta rutina?",
        "pt": "Em quais dias você quer fazer esta rotina?"
    },
    "routine_start_time": {
        "en": "When it will start?",
        "es": "¿Cuándo va a empezar?",
        "pt": "Quando vai começar?"
    },
    "routine_end_time": {
        "en": "When it will end?",
        "es": "¿Cuándo va a terminar?",
        "pt": "Quando vai terminar?"
    },
    "routine_delete_title": {
        "en": "Delete Routine",
        "es": "Eliminar Rutina",
        "pt": "Excluir Rotina"
    },
    "routine_delete_message": {
        "en": "Are you sure you want to delete this routine?",
        "es": "¿Estás seguro de que quieres eliminar esta rutina?",
        "pt": "Tem certeza que deseja excluir esta rotina?"
    },
    "routine_name_error": {
        "en": "Routine name is required",
        "es": "El nombre de la rutina es obligatorio",
        "pt": "O nome da rotina é obrigatório"
    },
    "routine_start_at": {
        "en": "Start at",
        "es": "Empieza a las",
        "pt": "Começa às"
    },
    "routine_end_at": {
        "en": "End at",
        "es": "Termina a las",
        "pt": "Termina às"
    },
    "screen_settings_language": {
        "en": "Language",
        "es": "Idioma",
        "pt": "Idioma"
    },
    "screen_settings_about": {
        "en": "About",
        "es": "Acerca de",
        "pt": "Sobre"
    },
    "screen_settings_clear_events": {
        "en": "Clear Events",
        "es": "Limpiar Eventos",
        "pt": "Limpar Eventos"
    }
}