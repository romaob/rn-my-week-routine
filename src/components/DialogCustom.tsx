import React from 'react';
import {View, StyleSheet, Modal} from 'react-native';
import Label, {FontSize} from './Label';
import Button, {ButtonColorType, ButtonSize} from './Button';
import {colors} from '../values/colors';
import {sizes} from '../values/sizes';
import {useString} from '../context/useStringContext';

interface DialogCustomProps {
  show: boolean;
  title: string;
  children: JSX.Element;
  cancelLabel?: string;
  confirmLabel?: string;
  onCancel?: () => void;
  onConfirm?: () => void;
}

const DialogCustom = ({
  show,
  title,
  children,
  cancelLabel,
  confirmLabel,
  onCancel,
  onConfirm,
}: DialogCustomProps): JSX.Element | null => {
  const {getString} = useString();
  if (!show) {
    return null;
  } else {
    return (
      <View style={styles.backdrop}>
        <Modal
          visible={show}
          animationType="fade"
          transparent
          testID="DialogCustom">
          <View style={styles.centeredView}>
            <View style={styles.container}>
              <View style={styles.topRow}>
                <Label
                  text={title}
                  size={FontSize.LARGE}
                  color={colors.light.primary}
                  testID="DialogCustomTitle"
                />
              </View>
              <View style={styles.body}>{children}</View>
              {(onCancel || onConfirm) && (
                <View style={styles.bottomRow}>
                  {onCancel && (
                    <Button
                      label={cancelLabel || getString('cancel')}
                      size={ButtonSize.LARGE}
                      onPress={onCancel}
                      colorType={ButtonColorType.SECONDARY}
                      testID="DialogCustomCancelButton"
                    />
                  )}
                  {onConfirm && (
                    <Button
                      label={confirmLabel || getString('ok')}
                      size={ButtonSize.LARGE}
                      onPress={onConfirm}
                      colorType={ButtonColorType.ACCENT}
                      testID="DialogCustomConfirmButton"
                    />
                  )}
                </View>
              )}
            </View>
          </View>
        </Modal>
      </View>
    );
  }
};

const styles = StyleSheet.create({
  backdrop: {
    display: 'flex',
    position: 'absolute',
    width: '120%',
    height: '120%',
    backgroundColor: 'rgba(0,0,0,0.5)',
    justifyContent: 'center',
    alignItems: 'center',
  },
  centeredView: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    display: 'flex',
  },
  container: {
    backgroundColor: colors.light.cardBackground,
    borderRadius: 10,
    padding: sizes.padding.lg,
    width: '80%',
  },
  topRow: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  body: {
    marginTop: 20,
    marginBottom: 20,
  },
  bottomRow: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  icon: {
    width: 20,
    height: 20,
  },
});

export default DialogCustom;
