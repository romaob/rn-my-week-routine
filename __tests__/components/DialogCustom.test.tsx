import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import DialogCustom from '../../src/components/DialogCustom';
import {Text} from 'react-native';

import strings from '../../src/values/strings.json';
import StringsProvider from '../../src/context/useStringContext';

const title = 'title';
const children = <Text testID="children">children</Text>;
const cancelLabel = 'cancelLabel';
const confirmLabel = 'confirmLabel';
const onCancel = jest.fn();
const onConfirm = jest.fn();

function testDialog(result: any) {
  expect(result).toBeDefined();
  expect(result.getByTestId('DialogCustom')).toBeDefined();
  expect(result.getByTestId('DialogCustomTitle')).toBeDefined();
  expect(result.getByText(title)).toBeDefined();
  expect(
    result
      .getByTestId('DialogCustomTitle')
      .props.children.some((child: any) => child === title),
  ).toBe(true);
  expect(result.getByTestId('children')).toBeDefined();
}

describe('DialogCustom render tests', () => {
  it('should not show the dialog when show is false', () => {
    const result = render(
      <DialogCustom show={false} title={title}>
        {children}
      </DialogCustom>,
    );
    expect(result.queryByTestId('DialogCustom')).toBeNull();
  });

  it('should render the dialog correctly when show is true', () => {
    const result = render(
      <DialogCustom
        show={true}
        title={title}
        cancelLabel={cancelLabel}
        confirmLabel={confirmLabel}
        onCancel={onCancel}
        onConfirm={onConfirm}>
        {children}
      </DialogCustom>,
    );
    expect(result).toMatchSnapshot();
    testDialog(result);
    expect(result.getByTestId('DialogCustomCancelButton')).toBeDefined();
    expect(result.getByText(cancelLabel)).toBeDefined();
    expect(result.getByTestId('DialogCustomConfirmButton')).toBeDefined();
    expect(result.getByText(confirmLabel)).toBeDefined();
  });

  it('should render the dialog with default buttons labels when not providing', () => {
    const result = render(
      <StringsProvider>
        <DialogCustom
          show={true}
          title={title}
          onCancel={onCancel}
          onConfirm={onConfirm}>
          {children}
        </DialogCustom>
      </StringsProvider>,
    );
    expect(result).toMatchSnapshot();
    testDialog(result);
    expect(result.getByTestId('DialogCustomCancelButton')).toBeDefined();
    expect(result.getByText(strings.cancel.en)).toBeDefined();
    expect(result.getByTestId('DialogCustomConfirmButton')).toBeDefined();
    expect(result.getByText(strings.cancel.en)).toBeDefined();
  });

  it('should not render the cancel button when onCancel is not provided', () => {
    const result = render(
      <DialogCustom
        show={true}
        title={title}
        cancelLabel={cancelLabel}
        confirmLabel={confirmLabel}
        onConfirm={onConfirm}>
        {children}
      </DialogCustom>,
    );
    expect(result).toMatchSnapshot();
    testDialog(result);
    expect(result.queryByTestId('DialogCustomCancelButton')).toBeNull();
    expect(result.getByTestId('DialogCustomConfirmButton')).toBeDefined();
    expect(result.getByText(confirmLabel)).toBeDefined();
  });

  it('should not render the confirm button when onConfirm is not provided', () => {
    const result = render(
      <DialogCustom
        show={true}
        title={title}
        cancelLabel={cancelLabel}
        confirmLabel={confirmLabel}
        onCancel={onCancel}>
        {children}
      </DialogCustom>,
    );
    expect(result).toMatchSnapshot();
    testDialog(result);
    expect(result.getByTestId('DialogCustomCancelButton')).toBeDefined();
    expect(result.getByText(cancelLabel)).toBeDefined();
    expect(result.queryByTestId('DialogCustomConfirmButton')).toBeNull();
  });
});

describe('DialogCustom behavior tests', () => {
  it('should call onCancel when cancel button is pressed', () => {
    const result = render(
      <DialogCustom
        show={true}
        title={title}
        cancelLabel={cancelLabel}
        confirmLabel={confirmLabel}
        onCancel={onCancel}
        onConfirm={onConfirm}>
        {children}
      </DialogCustom>,
    );
    expect(result).toMatchSnapshot();
    testDialog(result);
    expect(result.getByTestId('DialogCustomCancelButton')).toBeDefined();
    expect(result.getByText(cancelLabel)).toBeDefined();
    expect(result.getByTestId('DialogCustomConfirmButton')).toBeDefined();
    expect(result.getByText(confirmLabel)).toBeDefined();

    fireEvent.press(result.getByTestId('DialogCustomCancelButton'));
    expect(onCancel).toHaveBeenCalled();
  });

  it('should call onConfirm when confirm button is pressed', () => {
    const result = render(
      <DialogCustom
        show={true}
        title={title}
        cancelLabel={cancelLabel}
        confirmLabel={confirmLabel}
        onCancel={onCancel}
        onConfirm={onConfirm}>
        {children}
      </DialogCustom>,
    );
    expect(result).toMatchSnapshot();
    testDialog(result);
    expect(result.getByTestId('DialogCustomCancelButton')).toBeDefined();
    expect(result.getByText(cancelLabel)).toBeDefined();
    expect(result.getByTestId('DialogCustomConfirmButton')).toBeDefined();
    expect(result.getByText(confirmLabel)).toBeDefined();

    fireEvent.press(result.getByTestId('DialogCustomConfirmButton'));
    expect(onConfirm).toHaveBeenCalled();
  });
});
