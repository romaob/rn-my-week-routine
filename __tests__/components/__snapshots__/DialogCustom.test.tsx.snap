// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogCustom behavior tests should call onCancel when cancel button is pressed 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "rgba(0,0,0,0.5)",
      "display": "flex",
      "height": "120%",
      "justifyContent": "center",
      "position": "absolute",
      "width": "120%",
    }
  }
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    testID="DialogCustom"
    transparent={true}
    visible={true}
  >
    <View
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#ffffff",
            "borderRadius": 10,
            "padding": 24,
            "width": "80%",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              {
                "color": "#3a59db",
                "display": "flex",
                "flexDirection": "row",
                "fontSize": 24,
              }
            }
            testID="DialogCustomTitle"
          >
            title
          </Text>
        </View>
        <View
          style={
            {
              "marginBottom": 20,
              "marginTop": 20,
            }
          }
        >
          <Text
            testID="children"
          >
            children
          </Text>
        </View>
        <View
          style={
            {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#4791cb",
                "borderRadius": 5,
                "display": "flex",
                "justifyContent": "center",
                "opacity": 1,
                "padding": 8,
              }
            }
            testID="DialogCustomCancelButton"
          >
            <Text
              style={
                [
                  {
                    "color": "#ffffff",
                  },
                  {
                    "fontSize": 24,
                  },
                ]
              }
              testID="buttonLabel"
            >
              cancelLabel
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#d81e8b",
                "borderRadius": 5,
                "display": "flex",
                "justifyContent": "center",
                "opacity": 1,
                "padding": 8,
              }
            }
            testID="DialogCustomConfirmButton"
          >
            <Text
              style={
                [
                  {
                    "color": "#ffffff",
                  },
                  {
                    "fontSize": 24,
                  },
                ]
              }
              testID="buttonLabel"
            >
              confirmLabel
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;

exports[`DialogCustom behavior tests should call onConfirm when confirm button is pressed 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "rgba(0,0,0,0.5)",
      "display": "flex",
      "height": "120%",
      "justifyContent": "center",
      "position": "absolute",
      "width": "120%",
    }
  }
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    testID="DialogCustom"
    transparent={true}
    visible={true}
  >
    <View
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#ffffff",
            "borderRadius": 10,
            "padding": 24,
            "width": "80%",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              {
                "color": "#3a59db",
                "display": "flex",
                "flexDirection": "row",
                "fontSize": 24,
              }
            }
            testID="DialogCustomTitle"
          >
            title
          </Text>
        </View>
        <View
          style={
            {
              "marginBottom": 20,
              "marginTop": 20,
            }
          }
        >
          <Text
            testID="children"
          >
            children
          </Text>
        </View>
        <View
          style={
            {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#4791cb",
                "borderRadius": 5,
                "display": "flex",
                "justifyContent": "center",
                "opacity": 1,
                "padding": 8,
              }
            }
            testID="DialogCustomCancelButton"
          >
            <Text
              style={
                [
                  {
                    "color": "#ffffff",
                  },
                  {
                    "fontSize": 24,
                  },
                ]
              }
              testID="buttonLabel"
            >
              cancelLabel
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#d81e8b",
                "borderRadius": 5,
                "display": "flex",
                "justifyContent": "center",
                "opacity": 1,
                "padding": 8,
              }
            }
            testID="DialogCustomConfirmButton"
          >
            <Text
              style={
                [
                  {
                    "color": "#ffffff",
                  },
                  {
                    "fontSize": 24,
                  },
                ]
              }
              testID="buttonLabel"
            >
              confirmLabel
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;

exports[`DialogCustom render tests should not render the cancel button when onCancel is not provided 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "rgba(0,0,0,0.5)",
      "display": "flex",
      "height": "120%",
      "justifyContent": "center",
      "position": "absolute",
      "width": "120%",
    }
  }
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    testID="DialogCustom"
    transparent={true}
    visible={true}
  >
    <View
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#ffffff",
            "borderRadius": 10,
            "padding": 24,
            "width": "80%",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              {
                "color": "#3a59db",
                "display": "flex",
                "flexDirection": "row",
                "fontSize": 24,
              }
            }
            testID="DialogCustomTitle"
          >
            title
          </Text>
        </View>
        <View
          style={
            {
              "marginBottom": 20,
              "marginTop": 20,
            }
          }
        >
          <Text
            testID="children"
          >
            children
          </Text>
        </View>
        <View
          style={
            {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#d81e8b",
                "borderRadius": 5,
                "display": "flex",
                "justifyContent": "center",
                "opacity": 1,
                "padding": 8,
              }
            }
            testID="DialogCustomConfirmButton"
          >
            <Text
              style={
                [
                  {
                    "color": "#ffffff",
                  },
                  {
                    "fontSize": 24,
                  },
                ]
              }
              testID="buttonLabel"
            >
              confirmLabel
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;

exports[`DialogCustom render tests should not render the confirm button when onConfirm is not provided 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "rgba(0,0,0,0.5)",
      "display": "flex",
      "height": "120%",
      "justifyContent": "center",
      "position": "absolute",
      "width": "120%",
    }
  }
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    testID="DialogCustom"
    transparent={true}
    visible={true}
  >
    <View
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#ffffff",
            "borderRadius": 10,
            "padding": 24,
            "width": "80%",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              {
                "color": "#3a59db",
                "display": "flex",
                "flexDirection": "row",
                "fontSize": 24,
              }
            }
            testID="DialogCustomTitle"
          >
            title
          </Text>
        </View>
        <View
          style={
            {
              "marginBottom": 20,
              "marginTop": 20,
            }
          }
        >
          <Text
            testID="children"
          >
            children
          </Text>
        </View>
        <View
          style={
            {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#4791cb",
                "borderRadius": 5,
                "display": "flex",
                "justifyContent": "center",
                "opacity": 1,
                "padding": 8,
              }
            }
            testID="DialogCustomCancelButton"
          >
            <Text
              style={
                [
                  {
                    "color": "#ffffff",
                  },
                  {
                    "fontSize": 24,
                  },
                ]
              }
              testID="buttonLabel"
            >
              cancelLabel
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;

exports[`DialogCustom render tests should render the dialog correctly when show is true 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "rgba(0,0,0,0.5)",
      "display": "flex",
      "height": "120%",
      "justifyContent": "center",
      "position": "absolute",
      "width": "120%",
    }
  }
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    testID="DialogCustom"
    transparent={true}
    visible={true}
  >
    <View
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#ffffff",
            "borderRadius": 10,
            "padding": 24,
            "width": "80%",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              {
                "color": "#3a59db",
                "display": "flex",
                "flexDirection": "row",
                "fontSize": 24,
              }
            }
            testID="DialogCustomTitle"
          >
            title
          </Text>
        </View>
        <View
          style={
            {
              "marginBottom": 20,
              "marginTop": 20,
            }
          }
        >
          <Text
            testID="children"
          >
            children
          </Text>
        </View>
        <View
          style={
            {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#4791cb",
                "borderRadius": 5,
                "display": "flex",
                "justifyContent": "center",
                "opacity": 1,
                "padding": 8,
              }
            }
            testID="DialogCustomCancelButton"
          >
            <Text
              style={
                [
                  {
                    "color": "#ffffff",
                  },
                  {
                    "fontSize": 24,
                  },
                ]
              }
              testID="buttonLabel"
            >
              cancelLabel
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#d81e8b",
                "borderRadius": 5,
                "display": "flex",
                "justifyContent": "center",
                "opacity": 1,
                "padding": 8,
              }
            }
            testID="DialogCustomConfirmButton"
          >
            <Text
              style={
                [
                  {
                    "color": "#ffffff",
                  },
                  {
                    "fontSize": 24,
                  },
                ]
              }
              testID="buttonLabel"
            >
              confirmLabel
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;

exports[`DialogCustom render tests should render the dialog with default buttons labels when not providing 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "rgba(0,0,0,0.5)",
      "display": "flex",
      "height": "120%",
      "justifyContent": "center",
      "position": "absolute",
      "width": "120%",
    }
  }
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    testID="DialogCustom"
    transparent={true}
    visible={true}
  >
    <View
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#ffffff",
            "borderRadius": 10,
            "padding": 24,
            "width": "80%",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              {
                "color": "#3a59db",
                "display": "flex",
                "flexDirection": "row",
                "fontSize": 24,
              }
            }
            testID="DialogCustomTitle"
          >
            title
          </Text>
        </View>
        <View
          style={
            {
              "marginBottom": 20,
              "marginTop": 20,
            }
          }
        >
          <Text
            testID="children"
          >
            children
          </Text>
        </View>
        <View
          style={
            {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#4791cb",
                "borderRadius": 5,
                "display": "flex",
                "justifyContent": "center",
                "opacity": 1,
                "padding": 8,
              }
            }
            testID="DialogCustomCancelButton"
          >
            <Text
              style={
                [
                  {
                    "color": "#ffffff",
                  },
                  {
                    "fontSize": 24,
                  },
                ]
              }
              testID="buttonLabel"
            >
              Cancel
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#d81e8b",
                "borderRadius": 5,
                "display": "flex",
                "justifyContent": "center",
                "opacity": 1,
                "padding": 8,
              }
            }
            testID="DialogCustomConfirmButton"
          >
            <Text
              style={
                [
                  {
                    "color": "#ffffff",
                  },
                  {
                    "fontSize": 24,
                  },
                ]
              }
              testID="buttonLabel"
            >
              OK
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
